src/coreclr/vm/appdomainnative.cpp

```diff
@@ -38,7 +38,19 @@ void QCALLTYPE AppDomainNative::CreateDynamicAssembly(QCall::ObjectHandleOnStack
     args.access                 = access;
     args.stackMark              = stackMark;
 
-    Assembly *pAssembly = Assembly::CreateDynamic(GetAppDomain(), &args);
+    Assembly* pAssembly = nullptr;
+
+    if (assemblyLoadContext.Get() == NULL)
+    {
+        pAssembly = Assembly::CreateDynamic(GetAppDomain(), nullptr, &args);
+    }
+    else
+    {
+        INT_PTR         nativeAssemblyLoadContext = ((ASSEMBLYLOADCONTEXTREF)assemblyLoadContext.Get())->GetNativeAssemblyLoadContext();
+        ICLRPrivBinder* pBinderContext = reinterpret_cast<ICLRPrivBinder*>(nativeAssemblyLoadContext);
+
+        pAssembly = Assembly::CreateDynamic(GetAppDomain(), pBinderContext, &args);
+    }
```

Comment:
```suggestion
    ICLRPrivBinder* pBinderContext = nullptr;
    if (assemblyLoadContext.Get() != NULL)
    {
        INT_PTR         nativeAssemblyLoadContext = ((ASSEMBLYLOADCONTEXTREF)assemblyLoadContext.Get())->GetNativeAssemblyLoadContext();
        pBinderContext = reinterpret_cast<ICLRPrivBinder*>(nativeAssemblyLoadContext);
    }

    pAssembly = Assembly::CreateDynamic(GetAppDomain(), pBinderContext, &args);
```

It makes it easier to navigate the code if conceptually one callsite is really jus one callsite.